src/main/java/io/cryostat/net/web/http/api/v2/RuleDeleteHandler.java

```diff
@@ -162,7 +162,7 @@ public IntermediateResponse<Void> handle(RequestParameters params) throws ApiExc
     }
 
     private void cleanup(RequestParameters params, Rule rule) {
-        storage.listUniqueReachableServices().stream()
+        storage.listUniqueReachableServices().parallelStream()
```

Comment:
Just leave these as `stream()` calls, I think. The inner `vertx.executeBlocking` already moves the real work off onto a worker thread, so there is no need to use an implicit/hidden executor to try to do this step in parallel.